pkg/image/fetcher.go

```diff
@@ -123,6 +123,37 @@ func (f *Fetcher) Fetch(ctx context.Context, name string, options FetchOptions)
 	return f.fetchDaemonImage(name)
 }
 
+func (f *Fetcher) CheckReadAccessValidator(repo string, options FetchOptions) bool {
+	if !options.Daemon {
+		return f.checkRemoteReadAccess(repo)
+	}
+	if _, err := f.fetchDaemonImage(repo); err != nil {
+		// Image doesn't exist in the daemon
+		// 	Pull Never: should failed
+		// 	Pull Always or Pull If Not Present: need to check the registry
+		if options.PullPolicy == PullNever {
+			return false
+		}
+		return f.checkRemoteReadAccess(repo)
+	}
+	// no-op
+	return true
```

Comment:
But if `PullPolicy == PullAlways`, shouldn't we always check the remote access? Even if the image is readable in the daemon? Or would we just silently take the daemon one?

Reply:
> shouldn't we always check the remote access

I would think so

Reply:
I changed guard to be

```go
	if !options.Daemon || options.PullPolicy == PullAlways {
		return f.checkRemoteReadAccess(repo)
	}
```